### PR TITLE
MotC - Asahina Takamori

### LINKED ISSUE #2897: MotC - Asahina Takamori
image

### PR DESCRIPTION
**Pull request by jamesoddy**
Closes #2897

### CODE REVIEW

File: server/game/cards/08-MotC/AsahinaTakamori.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            },
+            target: {
+                controller: Players.Opponent,
+                cardType: CardTypes.Character,
+                cardCondition: (card, context) => card.getCost() <= context.event.card.getCost(),
+                gameAction: AbilityDsl.actions.cardLastingEffect({
+                    duration: Durations.UntilEndOfRound,
+                    effect: [
+                        AbilityDsl.effects.cannotParticipateAsAttacker(),
+                        AbilityDsl.effects.cannotParticipateAsDefender()
```
Comment (Kwaice) [inline server/game/cards/08-MotC/AsahinaTakamori.js]: I think it should be declareAsAttacker here.

Comment (jamesoddy) [reply server/game/cards/08-MotC/AsahinaTakamori.js]: Ah good call.  Fix incoming.

Comment (jamesoddy) [reply server/game/cards/08-MotC/AsahinaTakamori.js]: Huh, I thought we had that as an effect already, but doesn't look like it.  I'll have to write a new one :)

Comment (jamesoddy) [reply server/game/cards/08-MotC/AsahinaTakamori.js]: Nevermind, found it as a `cardCannot` effect

File: server/game/cards/08-MotC/AsahinaTakamori.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const DrawCard = require('../../drawcard.js');
+const { CardTypes, Durations, Players } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl');
+
+class AsahinaTakamori extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Choose a character to pacify',
```
Comment (jeremylarner) [inline server/game/cards/08-MotC/AsahinaTakamori.js]: This is a title, not a prompt, so I would make it 'Pacify a character'

File: server/game/cards/08-MotC/AsahinaTakamori.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            when: {
+                onCardPlayed: (event) => event.card.type === CardTypes.Character && event.card.isFaction('crane')
```
Comment (jeremylarner) [inline server/game/cards/08-MotC/AsahinaTakamori.js]: You need to check that `context.player` is playing the card

File: server/game/cards/08-MotC/AsahinaTakamori.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            },
+            target: {
+                controller: Players.Opponent,
+                cardType: CardTypes.Character,
+                cardCondition: (card, context) => card.getCost() <= context.event.card.getCost(),
```
Comment (jeremylarner) [inline server/game/cards/08-MotC/AsahinaTakamori.js]: You should use `card.costLessThan(num)` to deal with cards with no cost being immune to these effects

File: server/game/cards/08-MotC/AsahinaTakamori.js (hunk 3 above)
Comment (jamesoddy) [reply server/game/cards/08-MotC/AsahinaTakamori.js]: Good catch.  I've added a test and fix for that.

File: server/game/cards/08-MotC/AsahinaTakamori.js (hunk 4 above)
Comment (jamesoddy) [reply server/game/cards/08-MotC/AsahinaTakamori.js]: Changed.

**Comment by gryffon** [comment]
Looks like this has some test failures.

**Pull request merged**